### ISSUE TITLE
[vcpkg scripts] CMake policy CMP0174 warning from every package with CMake 3.31

### ISSUE THREAD

**Issue by ADKaster** [microsoft/vcpkg#41999]
**Describe the bug**
With CMake 3.31, every call to z_vcpkg_find_acquire_program_find_external or z_vcpkg_find_acquire_program_find_internal is giving a dev warning from one of two places: line 70 or line 30

**Environment**
- OS: Ubuntu 24.04
- Compiler: clang-18
- CMake: 3.31 (from kitware apt repos)

**To Reproduce**
0. install cmake 3.31+
1. ./vcpkg install woff2
2. See dev warnings

```
CMake Warning (dev) at scripts/cmake/vcpkg_find_acquire_program.cmake:70 (cmake_parse_arguments):
  The INTERPRETER keyword was followed by an empty string or no value at all.
  Policy CMP0174 is not set, so cmake_parse_arguments() will unset the
  arg_INTERPRETER variable rather than setting it to an empty string.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_find_acquire_program.cmake:143 (z_vcpkg_find_acquire_program_find_internal)
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:193 (vcpkg_find_acquire_program)
  buildtrees/versioning_/versions/woff2/fbd2207ad0ad1e7e52b421b86f96deff9ed7ec65/portfile.cmake:33 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:192 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Installing: /home/andrew/ladybird-org/ladybird-browser/Build/vcpkg/packages/woff2_x64-linux/share/woff2/copyright
CMake Warning (dev) at scripts/cmake/vcpkg_find_acquire_program.cmake:70 (cmake_parse_arguments):
  The INTERPRETER keyword was followed by an empty string or no value at all.
  Policy CMP0174 is not set, so cmake_parse_arguments() will unset the
  arg_INTERPRETER variable rather than setting it to an empty string.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_find_acquire_program.cmake:143 (z_vcpkg_find_acquire_program_find_internal)
  scripts/cmake/z_vcpkg_fixup_rpath.cmake:96 (vcpkg_find_acquire_program)
  scripts/ports.cmake:202 (z_vcpkg_fixup_rpath_in_dir)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at scripts/cmake/vcpkg_find_acquire_program.cmake:30 (cmake_parse_arguments):
  The INTERPRETER keyword was followed by an empty string or no value at all.
  Policy CMP0174 is not set, so cmake_parse_arguments() will unset the
  arg_INTERPRETER variable rather than setting it to an empty string.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_find_acquire_program.cmake:149 (z_vcpkg_find_acquire_program_find_external)
  scripts/cmake/z_vcpkg_fixup_rpath.cmake:96 (vcpkg_find_acquire_program)
  scripts/ports.cmake:202 (z_vcpkg_fixup_rpath_in_dir)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Expected behavior**
No warnings from vcpkg internals.

I would still like to have CMake dev warnings for my own scripts in my project, so `-Wno-dev` doesn't seem appealing

**Failure logs**
N/A

**Additional context**
None

**Comment by dg0yt** [comment]
Is there a release of 3.31? This issues was observed in the release candidates.

**Comment by dg0yt** [comment]
See https://gitlab.kitware.com/cmake/cmake/-/issues/26397.

**Comment by ADKaster** [comment]
3.31 was tagged today, Wed Nov 6. https://gitlab.kitware.com/cmake/cmake/-/commits/v3.31.0?ref_type=tags

**Comment by dg0yt** [comment]
But which version did you actual test?

**Comment by dg0yt** [comment]
The fix is trivial, but it invalidates most cached binary artifacts (aka world rebuild).

**Comment by jeanga** [comment]
> But which version did you actual test?

It reproes with the released version of cmake 3.31 (available via winget)

**Comment by ADKaster** [comment]
> But which version did you actual test?

```
$ cmake --version
cmake version 3.31.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

From https://apt.kitware.com/
